src/app/blog/[slug]/page.tsx

```diff
@@ -1,49 +1,53 @@
 import React from "react";
-import fs from "fs";
 import md from "markdown-it";
 import { getPostBySlug } from "../utils";
 import { type BlogWriter } from "../writers";
 import Image from "next/image";
 import Link from "next/link";
+import { notFound } from "next/navigation";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { faUserCircle } from "@fortawesome/free-solid-svg-icons";
 
 // NOTE: https://blog.openreplay.com/creating-a-markdown-blog-powered-by-next-js-in-under-an-hour
 
 function AuthorTile(props: { author: BlogWriter }): JSX.Element {
   const { author } = props;
   return (
-    <div className="flex flex-col md:flex-row">
-      <div className="h-32 w-32 mb-6 md:w-96 relative mr-5">
-        <Image
-          src={author.profile_picture_url}
-          alt="Blog Image"
-          className="object-cover w-full h-full rounded-full"
-          sizes="100vw"
-          fill
-        />
+    <div className="flex flex-col mt-10 mb-6 gap-4 md:flex-row md:items-center">
+      <div className="aspect-square h-24 w-24 md:h-32 md:w-32 relative">
+        {author.profile_picture_url !== "" ? (
+          <Image
+            src={author.profile_picture_url}
+            alt="Blog Image"
+            className="object-cover object-center w-full h-full rounded-full m-0"
+            sizes="100vw"
+            fill
+          />
+        ) :
+          <FontAwesomeIcon
+            className="object-cover object-center w-full h-full rounded-full m-0"
+            color="#DDD"
+            icon={faUserCircle}
+          />
+        }
       </div>
       <div>
-        <p className="mb-2 font-bold"> {author.full_name} </p>
-        <p className="mt-0"> {author.description} </p>
+        <p className="m-0 mb-2 font-bold"> {author.full_name} </p>
+        <p className="m-0"> {author.description} </p>
       </div>
     </div>
   );
 }
 
-// TODO Type this properly
-export async function generateStaticParams(): Promise<any> {
```

Comment:
Im still undecided whether removing this is the right plan. 

The nice thing about it is we can render blogs as soon as they get released on the website. If we keep this then we have to do some sort of build everytime we release a blog which is a bit of a pain.

The annoying thing is
1. We have to fetch the blog each fetch (we can solve this with some caching below) but this is a very small issue
2. If we are going to build a site map (which it sounds like we want) then for blogs I assume we will want to know all of them at build time (so we can still all the blogs in the sitemap?)

What do you think makes sense?

Reply:
I suppose for now we can do it in two ways. One is to fetch everything dynamically (which is basically what this pull request does). I assume you worried about too much API calls right? We can definitely reduce API calls, by:

1. having next.js cache accessed webpages (request results) for a fixed period (see https://nextjs.org/docs/app/api-reference/functions/fetch#optionsnextrevalidate), set it to cache blog posts for a day seems reasonable to me (which is not added to this pull request yet), so that the number of API calls = the number of times Next.js request new data for cache purposes.
2. Same thing applies to sitemaps: we can have next.js dynamically generate a new site map after a fixed period passed.

I think this is the better way to do it. We can do some tests to see if the backend calls reduced after we added dynamic content cache to our project.

As of the second approach: we can also do it fully statically by only fetching data once at build time. I imagine you can add something (called webhooks?) to the backend so that any update to the blogs (or any other part of the backend data) to trigger a new frontend build, which should work just fine if we don't update blogs very frequently. The benefit of this is that we can render the whole website as a static website rather than server-side rendering, in theory because we can distribute static content via CDN the performance would be better. Then again, it will probably take more time to build considering this is an architecture change. 

Let me know if there is anything need to be clarified.